Jellyfin.Server/Program.cs

```diff
@@ -242,9 +242,29 @@ private static ServerApplicationPaths CreateApplicationPaths(StartupOptions opti
                 }
             }
 
+            // Ensure the three main folders exist before we continue
+            EnsureCreated(dataDir);
+            EnsureCreated(configDir);
+            EnsureCreated(logDir);
+
             return new ServerApplicationPaths(dataDir, logDir, configDir);
         }
 
+        private static void EnsureCreated(string path)
+        {
+            try
+            {
+                Directory.CreateDirectory(path);
+            }
+            catch (Exception ex)
+            {
+                var errorWriter = Console.Error;
+                errorWriter.WriteLine("Error whilst attempting to create program data folder: {0}", path);
```

Comment:
Can we add an arg with the name of folder? So we can tell user "problems creating log dir".

Reply:
We sure can.